### PR TITLE
local screens and screen values

### LINKED ISSUE #184: Local screens
Create local screen definitions from scripts and have them open in another TlmViewer tab

### PR DESCRIPTION
**Pull request by ryanmelt**
closes #184

### CODE REVIEW

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue [hunk 1]
```diff
@@ -196,10 +196,6 @@ export default {
       dragHandle: '.v-system-bar',
     })
     this.grid.on('dragEnd', this.refreshLayout)
-    const previousConfig = localStorage['lastconfig__telemetry_viewer']
-    if (previousConfig) {
-      this.openConfiguration(previousConfig)
-    }
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-cosmos-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue:202]: Why are you removing this?

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/Openc3Screen.vue [hunk 2]
```diff
@@ -720,6 +724,37 @@ export default {
...
+              message: error.message,
+              time: new Date().getTime(),
+            })
+          })
+      }
+    },
+    updateValues: function (values) {
+      this.updateCounter += 1
+      for (let i = 0; i < values.length; i++) {
+        values[i].push(this.updateCounter)
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/Openc3Screen.vue:746]: What is the point of this updateCounter? It looks like it gets set to 0 initially and then just increments forever?

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/CanvasimagevalueWidget.vue [hunk 3]
```diff
@@ -56,44 +56,43 @@ export default {
...
     selectedValue: function () {
-      return this.$store.state.tlmViewerValues[this.valueId][0]
+      if (this.screen) {
+        if (this.screen.screenValues[this.valueId]) {
+          return this.screen.screenValues[this.valueId][0]
+        }
+      } else if (this.$store.state.tlmViewerValues[this.valueId]) {
+        return this.$store.state.tlmViewerValues[this.valueId][0]
+      }
+      return null
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/CanvasimagevalueWidget.vue:72]: Don't all widgets have a screen? When is the else case used?

File: openc3/lib/openc3/script/screen.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      # Noop outside of ScriptRunner
+    end
+
+    def clear_all_screens
+      # Noop outside of ScriptRunner
+    end
+
+    def local_screen(screen_name, definition, x = nil, y = nil)
+      # Noop outside of ScriptRunner
+    end
```
Comment (jmthomas) [inline openc3/lib/openc3/script/screen.rb:117]: Love to see a new script in the DEMO called `screens.rb` that demonstrates all this.

**Review by jmthomas — APPROVED** [review]
When you create a local screen does it get persisted?

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue (hunk 1 above)
Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-cosmos-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue:202]: This was really bugging me.  I didn't want the the same config to keep coming up and there is no built in method to stop it.   Needs to be redone with some kind of pin on/off.

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/Openc3Screen.vue (hunk 2 above)
Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/Openc3Screen.vue:746]: I believe it drives the aging logic

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/CanvasimagevalueWidget.vue (hunk 3 above)
Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/CanvasimagevalueWidget.vue:72]: We can probably eliminate the $store values but that will be a large refactor.  Let's talk tomorrow.

File: openc3/lib/openc3/script/screen.rb (hunk 4 above)
Comment (ryanmelt) [reply openc3/lib/openc3/script/screen.rb:117]: Done

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/CanvasimagevalueWidget.vue (hunk 3 above)
Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/CanvasimagevalueWidget.vue:70]: Eleminated $store and opened a new ticket for further refactoring

**Review by jmthomas — APPROVED** [review]
I was expecting all the `if (this.screen)` code to also go away but I guess better safe than sorry. Don't see how that check could ever fail now though.

**Pull request merged**